### PR TITLE
Remove compatibility for flag tracking_records_in_ets

### PR DESCRIPTION
**Pull request by dcorbacho**
Remove compatibility code for feature flag `tracking_records_in_ets` because this feature flag is required in 3.12.

See https://github.com/rabbitmq/rabbitmq-server/pull/7219

### CODE REVIEW

File: deps/rabbit/src/rabbit_channel_tracking.erl [hunk 1]
```diff
@@ -116,27 +110,14 @@ handle_cast({user_deleted, Details}) ->
...
-        false ->
-            Node = pget(node, Details),
-            rabbit_log_channel:info(
-              "Node '~ts' was removed from the cluster, deleting"
-              " its channel tracking tables...", [Node]),
-            delete_tracked_channels_table_for_node(Node),
-            delete_per_user_tracked_channels_table_for_node(Node)
-    end.
+handle_cast({node_deleted, _Details}) ->
+    ok.
```
Comment (ansd) [inline deps/rabbit/src/rabbit_channel_tracking.erl:114]: It looks like we can delete this `handle_cast/1` clause altogether and also delete https://github.com/rabbitmq/rabbitmq-server/blob/7606bcfd17202a3a42c4a012a41c2a6f31f57fa3/deps/rabbit/src/rabbit_channel_tracking_handler.erl#L54-L57

File: deps/rabbit/src/rabbit_channel_tracking.erl [hunk 2]
```diff
@@ -116,27 +110,14 @@ handle_cast({user_deleted, Details}) ->
...
 
 -spec register_tracked(rabbit_types:tracked_channel()) -> ok.
 -dialyzer([{nowarn_function, [register_tracked/1]}]).
 
 register_tracked(TrackedCh = #tracked_channel{node = Node}) when Node == node() ->
-    case rabbit_feature_flags:is_enabled(tracking_records_in_ets) of
-        true -> register_tracked_ets(TrackedCh);
-        false -> register_tracked_mnesia(TrackedCh)
-    end.
+    register_tracked_ets(TrackedCh).
```
Comment (ansd) [inline deps/rabbit/src/rabbit_channel_tracking.erl:120]: Let's delete `register_tracked_ets/1` and inline its function body directly into `register_tracked/1`

File: deps/rabbit/src/rabbit_channel_tracking.erl [hunk 3]
```diff
@@ -149,27 +130,9 @@ register_tracked_ets(TrackedCh = #tracked_channel{pid = ChPid, username = Userna
...
-    end,
-    ok.
-
 -spec unregister_tracked_by_pid(pid()) -> any().
 unregister_tracked_by_pid(ChPid) when node(ChPid) == node() ->
-    case rabbit_feature_flags:is_enabled(tracking_records_in_ets) of
-        true -> unregister_tracked_by_pid_ets(ChPid);
-        false -> unregister_tracked_by_pid_mnesia(ChPid)
-    end.
+    unregister_tracked_by_pid_ets(ChPid).
```
Comment (ansd) [inline deps/rabbit/src/rabbit_channel_tracking.erl:135]: Let's delete `unregister_tracked_by_pid_ets/1` and inline its function body directly into `unregister_tracked_by_pid/1`

File: deps/rabbit/src/rabbit_channel_tracking.erl [hunk 4]
```diff
@@ -179,27 +142,13 @@ unregister_tracked_by_pid_ets(ChPid) ->
...
 %% callback, however it is not used in rabbitmq-server any more. It is
 %% only kept for backwards compatibility if 3rd-party code would rely
 %% on it.
 -spec unregister_tracked(rabbit_types:tracked_channel_id()) -> ok.
 unregister_tracked(ChId = {Node, _Name}) when Node == node() ->
-    case rabbit_feature_flags:is_enabled(tracking_records_in_ets) of
-        true -> unregister_tracked_ets(ChId);
-        false -> unregister_tracked_mnesia(ChId)
-    end.
+    unregister_tracked_ets(ChId).
```
Comment (ansd) [inline deps/rabbit/src/rabbit_channel_tracking.erl:151]: Same here.

File: deps/rabbit/src/rabbit_channel_tracking.erl [hunk 5]
```diff
@@ -209,23 +158,10 @@ unregister_tracked_ets(ChId) ->
...
-    end.
-
 -spec count_tracked_items_in({atom(), rabbit_types:username()}) -> non_neg_integer().
 
 count_tracked_items_in(Type) ->
-    case rabbit_feature_flags:is_enabled(tracking_records_in_ets) of
-        true -> count_tracked_items_in_ets(Type);
-        false -> count_tracked_items_in_mnesia(Type)
-    end.
+    count_tracked_items_in_ets(Type).
```
Comment (ansd) [inline deps/rabbit/src/rabbit_channel_tracking.erl:164]: Same here.

File: deps/rabbit/src/rabbit_channel_tracking.erl [hunk 6]
```diff
@@ -236,19 +172,10 @@ count_tracked_items_in_ets({user, Username}) ->
...
-        #tracked_channel_per_user.channel_count, Username,
-        "channels of user").
-
 -spec clear_tracking_tables() -> ok.
 
 clear_tracking_tables() ->
-    case rabbit_feature_flags:is_enabled(tracking_records_in_ets) of
-        true -> ok;
-        false -> clear_tracked_channel_tables_for_this_node()
-    end.
```
Comment (ansd) [inline deps/rabbit/src/rabbit_channel_tracking.erl:251]: Can we now delete function `clear_tracking_tables/0` and also delete its callback declaration in `rabbit_tracking` ?

File: deps/rabbit/src/rabbit_channel_tracking.erl [hunk 7]
```diff
@@ -267,53 +194,27 @@ list() ->
 -spec list_of_user(rabbit_types:username()) -> [rabbit_types:tracked_channel()].
 
 list_of_user(Username) ->
-    case rabbit_feature_flags:is_enabled(tracking_records_in_ets) of
-        true -> list_of_user_ets(Username);
-        false -> list_of_user_mnesia(Username)
-    end.
+    list_of_user_ets(Username).
```
Comment (ansd) [inline deps/rabbit/src/rabbit_channel_tracking.erl:197]: Same here.

**Review by ansd — CHANGES_REQUESTED** [review]

File: deps/rabbit/src/rabbit_connection_tracking.erl [hunk 8]
```diff
@@ -150,26 +141,14 @@ handle_cast({user_deleted, Details}) ->
...
-            ok;
-        false ->
-            Node = pget(node, Details),
-            rabbit_log_connection:info("Node '~ts' was removed from the cluster, deleting its connection tracking tables...", [Node]),
-            delete_tracked_connections_table_for_node(Node),
-            delete_per_vhost_tracked_connections_table_for_node(Node),
-            delete_per_user_tracked_connections_table_for_node(Node)
-    end.
+handle_cast({node_deleted, _Details}) ->
+    ok.
```
Comment (ansd) [inline deps/rabbit/src/rabbit_connection_tracking.erl:145]: these two lines can be deleted

File: deps/rabbit/src/rabbit_connection_tracking.erl [hunk 9]
```diff
@@ -450,18 +264,11 @@ lookup(Name, [Node | Nodes]) ->
     end.
 
 lookup_internal(Name, Node) ->
-    case rabbit_feature_flags:is_enabled(tracking_records_in_ets) of
-        true -> lookup_ets(Name, Node);
-        false -> lookup_mnesia(Name, Node)
-    end.
+    lookup_ets(Name, Node).
```
Comment (ansd) [inline deps/rabbit/src/rabbit_connection_tracking.erl:267]: There are multiple functions in this file which just call `*_ets`.
I think they can be all inlined meaning the `*_ets` functions can be deleted.
WDYT?

**Review by ansd — CHANGES_REQUESTED** [review]

File: deps/rabbit/src/rabbit_connection_tracking.erl (hunk 9 above)
Comment (dcorbacho) [reply deps/rabbit/src/rabbit_connection_tracking.erl]: Done

File: deps/rabbit/src/rabbit_connection_tracking.erl [hunk 10]
```diff
@@ -68,10 +62,6 @@
 %% Sets up and resets connection tracking tables for this
 %% node.
 boot() ->
-    ensure_tracked_connections_table_for_this_node(),
-    ensure_per_vhost_tracked_connections_table_for_this_node(),
-    ensure_per_user_tracked_connections_table_for_this_node(),
-    clear_tracking_tables(),
     ok.
```
Comment (ansd) [inline deps/rabbit/src/rabbit_connection_tracking.erl:65]: It looks like we should also delete the following `rabbit` boot steps:
https://github.com/rabbitmq/rabbitmq-server/blob/56e4ed5464b3f23b682639378c4cab58879c288a/deps/rabbit/src/rabbit.erl#L212-L220

and delete functions `rabbit_connection_tracking:boot/0` and `rabbit_channel_tracking:boot/0` and the callback definition in https://github.com/rabbitmq/rabbitmq-server/blob/56e4ed5464b3f23b682639378c4cab58879c288a/deps/rabbit/src/rabbit_tracking.erl#L16

**Review by ansd — APPROVED** [review]

**Pull request merged**